varats/tools/research_tools/vara.py

```diff
@@ -0,0 +1,202 @@
+    optinaly phasar, for static analysis.
+    """
+
+    def __init__(self, base_dir: Path) -> None:
+        sub_projects = [
+            SubProject("vara-llvm-project",
+                       "https://github.com/llvm/llvm-project.git", "upstream",
+                       "vara-llvm-project"),
+            SubProject("VaRA", "git@github.com:se-passau/VaRA.git", "origin",
+                       "vara-llvm-project/vara"),
+            SubProject(
+                "phasar",
+                "https://github.com/secure-software-engineering/phasar.git",
+                "origin", "vara-llvm-project/phasar")
+        ]
+        super().__init__(base_dir, sub_projects)
+
+    def setup_vara_remotes(self) -> None:
+        """
+        Sets up VaRA specific upstream remotes for projects that where forked.
+        """
+        self.get_sub_project("vara-llvm-project").add_remote(
+            self.base_dir, "origin",
+            "git@github.com:se-passau/vara-llvm-project.git")
+
+    def setup_build_link(self) -> None:
+        """
+        Setup build-config folder link for VaRAs default build setup scripts.
+        """
+        llvm_project_dir = self.base_dir / self.get_sub_project(
+            "vara-llvm-project").path
+        mkdir(llvm_project_dir / "build/")
+        with local.cwd(llvm_project_dir / "build/"):
+            ln("-s", llvm_project_dir / "vara/utils/vara/builds/", "build_cfg")
+
+    def checkout_vara_version(self, version: int,
+                              use_dev_branches: bool) -> None:
+        """
+        Checkout out a specific version of VaRA.
+
+        Args:
+            version: major version number, e.g., 100 or 110
+            use_dev_branches: true, if one wants the current development version
+        """
+        dev_suffix = "-dev" if use_dev_branches else ""
+        LOG.info(f"Checking out VaRA version {version}" + dev_suffix)
+
+        self.get_sub_project("vara-llvm-project").checkout_branch(
+            self.base_dir, f"vara-{version}" + dev_suffix)
+
+        # TODO (sattlerf): make different checkout for older versions
+        self.get_sub_project("VaRA").checkout_branch(self.base_dir,
+                                                     f"vara" + dev_suffix)
+        self.get_sub_project("phasar").checkout_branch(self.base_dir,
+                                                       "development")
+
+    def setup_submodules(self) -> None:
+        """
+        Set up the git submodules of all sub projects.
+        """
+        self.get_sub_project("phasar").init_and_update_submodules(self.base_dir)
+
+
+class VaRA(ResearchTool[VaRACodeBase]):
+    """
+    Research tool implementation for VaRA.
+    Find the main repo online on github: https://github.com/se-passau/VaRA
+    """
+
+    def __init__(self, base_dir: Path) -> None:
+        super().__init__("VaRA", [BuildType.DEV], VaRACodeBase(base_dir))
+
+    def setup(self, source_folder: Path, **kwargs: tp.Any) -> None:
```

Comment:
```suggestion
    @check_required_args(["install_prefix"])
    def setup(self, source_folder: Path, **kwargs: tp.Any) -> None:
```

Don't forget the import.

Also, maybe `plot_utils` isn't the best location for that decorator.

Reply:
Yeah, we should consider moving it at some point.
I also had the idea of making the implementation use introspection to figure out what is requested from 'kwargs', so the user does not have to provide the list.